homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -159,6 +164,10 @@
         "method_on": "async_set_clean_on",
         "method_off": "async_set_clean_off",
     },
+    SERVICE_SET_LED: {
```

Comment:
It's confusing to call this service when it's not a separate service.

Reply:
But we don't register `set_led` as a service. We use this to call appropriate method here:
https://github.com/home-assistant/core/blob/94d1c7ae3a8e6d27bd919d7a82b0beff570cec56/homeassistant/components/xiaomi_miio/switch.py#L440-L456

Reply:
Services mentions here are refering to backend services. WIth the new impementation we remove the custom integration specific sevices and make use of the generic platform services.

Reply:
Right, that's how it's used for the coordinator entity. But a service is also registered in the other entry setup. And that service will basically be a noop.

Reply:
The coordinator entity provides the sync with the backend state. Where changes are, like a switch toggle is perform by the user, the change will be changed offline form the backend state. This avoids unneeded io to the backend.

Reply:
@jbouwh I'm talking about the services that are registered for the other entry that doesn't use a coordinator. It's using this same constant to do that. It's confusing and incorrect to mix these two different cases.

Reply:
> 
> 
> @jbouwh I'm talking about the services that are registered for the other entry that doesn't use a coordinator. It's using this same constant to do that. It's confusing and incorrect to mix these two different cases.

We have made an special class for coordinator based devices here (`XiaomiGenericCoordinatedSwitch`) and will migrate all other legacy classes to use the coordinator.

Reply:
I used `SwitchEntityDescription` class and removed "coordinated" switches from the `SERVICE_TO_METHOD` constants. What do you think?

Reply:
Nice!